Smart handling of processes leftovers in tests

azat opened the pull request:
Previously processes cleanup on i.e. SIGINT simply did not work, because
the launcher kills only processes in process group, while tests are
launched with start_new_session=True for Popen(), which creates own
process group.

This is needed for killing process group in case of test timeout.

So instead, look at the parent pid, and **kill the child process groups**.

Also add some **logging to make it more informative** which processes will be killed.

**And now in case of test timeout `clickhouse-test` will gather stacktraces from clickhouse utilities as well** (by sending SIGTSTP to them), this should help with issues like https://github.com/ClickHouse/ClickHouse/issues/67736

And also **omit python stacktraces** in some cases when it is useless and only create noise (server died, clickhouse-test signals, i.e. the most common is SIGINT - **Ctrl-C**)

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

azat committed:
tests: avoid leaving processes leftovers

Previously processes cleanup on i.e. SIGINT simply did not work, because
the launcher kills only processes in process group, while tests are
launched with start_new_session=True for Popen(), which creates own
process group.

This is needed for killing process group in case of test timeout.

So instead, look at the parent pid, and kill the child process groups.

Also add some logging to make it more explicit which processes will be
killed.

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests: omit python stacktace in case of signals/server died

It is simply useless and only create output that only distracts.

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests: try to catch stacktraces from client in case of test timeouts

This is to catch issues like [1].

  [1]: https://github.com/ClickHouse/ClickHouse/issues/67736

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests: capture stderr/stdout/debuglog after terminating test

It was simply wrong before, but now, with capturing stacktrace that can
take sometime it is a must.

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

robot-clickhouse-ci-2 commented:
<!-- automatic status comment for PR #67737 from azat/ClickHouse:tests-processes-leftovers -->
*This is an automated comment for commit ffca4b8093906152d767626a1e239a75a71e5f92 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67737/ffca4b8093906152d767626a1e239a75a71e5f92/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ddeb4e7b5572783b72292c1c51c6e9308e9ca16c/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

azat committed:
tests: fix pylint issue in clickhouse_execute_http()

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented:
>Integration tests (asan, old analyzer) [4/6] — fail: 48, passed: 462

- https://github.com/ClickHouse/ClickHouse/pull/67743

alexey-milovidov commented:
@azat Merge with master to include the fixes.

azat commented:
Done

azat commented:
>Stress test (asan) — check_status.tsv doesn't exists

```
Uexpected exception, will not retry:  HTTPError :  Code: 503. Code: 439. DB::Exception: Cannot schedule a task: cannot allocate thread (threads=1, jobs=1). (CANNOT_SCHEDULE_TASK) (version 24.8.1.1342)
```

- https://github.com/ClickHouse/ClickHouse/issues/67775

>Integration tests (aarch64) [4/6] — fail: 1, passed: 400

- `test_executable_user_defined_function/test.py::test_executable_function_slow_python`

azat commented:
Yeah, it could kill more then it should:

```
Killing process group 1
Processes in process group 1:
1 /bin/bash /run.sh
276 ./minio server --address :11111 ./minio_data
4391 gdb -batch -command script.gdb -p 3556
4392 /usr/bin/perl /usr/bin/ts %Y-%m-%d %H:%M:%S
4443 python3 /usr/bin/stress --hung-check --drop-databases --output-folder test_output --skip-func-tests  --global-time-limit 1200
4460 /bin/sh -c /usr/bin/clickhouse-test --order=random  --database=test_1  --client-option join_use_nulls=1 join_algorithm='parallel_hash' memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 group_by_use_nulls=1 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas='parallel_replicas' parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200  
4462 python3 /usr/bin/clickhouse-test --order=random --database=test_1 --client-option join_use_nulls=1 join_algorithm=parallel_hash memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 group_by_use_nulls=1 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas=parallel_replicas parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200
4495 /bin/sh -c /usr/bin/clickhouse-test --order=random  --client-option join_use_nulls=1 http_make_head_request=0 ignore_drop_queries_probability=0.5 --global_time_limit=1200  
4500 python3 /usr/bin/clickhouse-test --order=random --client-option join_use_nulls=1 http_make_head_request=0 ignore_drop_queries_probability=0.5 --global_time_limit=1200
4530 /bin/sh -c /usr/bin/clickhouse-test --order=random  --client-option memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas='parallel_replicas' parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200  
4534 python3 /usr/bin/clickhouse-test --order=random --client-option memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas=parallel_replicas parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200
4550 /bin/sh -c /usr/bin/clickhouse-test --order=random  --database=test_7  --client-option join_use_nulls=1 join_algorithm='grace_hash' group_by_use_nulls=1 ignore_drop_queries_probability=0.5 --global_time_limit=1200  
4552 python3 /usr/bin/clickhouse-test --order=random --database=test_7 --client-option join_use_nulls=1 join_algorithm=grace_hash group_by_use_nulls=1 ignore_drop_queries_probability=0.5 --global_time_limit=1200
4643 /bin/sh -c /usr/bin/clickhouse-test --order=random  --client-option implicit_transaction=1 throw_on_unsupported_query_inside_transaction=0 ignore_drop_queries_probability=0.5 --global_time_limit=1200  
4648 python3 /usr/bin/clickhouse-test --order=random --client-option implicit_transaction=1 throw_on_unsupported_query_inside_transaction=0 ignore_drop_queries_probability=0.5 --global_time_limit=1200
4772 /bin/sh -c /usr/bin/clickhouse-test --order=random  --client-option join_use_nulls=1 memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 http_make_head_request=0 ignore_drop_queries_probability=0.5 --global_time_limit=1200  
4782 python3 /usr/bin/clickhouse-test --order=random --client-option join_use_nulls=1 memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 http_make_head_request=0 ignore_drop_queries_probability=0.5 --global_time_limit=1200
4812 /bin/sh -c /usr/bin/clickhouse-test --order=random  --client-option optimize_trivial_approximate_count_query=1 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas='parallel_replicas' parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200  
4825 python3 /usr/bin/clickhouse-test --order=random --client-option optimize_trivial_approximate_count_query=1 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas=parallel_replicas parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200
4924 /bin/sh -c /usr/bin/clickhouse-test --order=random  --database=test_21  --client-option join_algorithm='parallel_hash' memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 group_by_use_nulls=1 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas='parallel_replicas' parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200  
4947 python3 /usr/bin/clickhouse-test --order=random --database=test_21 --client-option join_algorithm=parallel_hash memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 group_by_use_nulls=1 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas=parallel_replicas parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200
5033 /bin/sh -c /usr/bin/clickhouse-test --order=random  --client-option http_make_head_request=0 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas='parallel_replicas' parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200  
5036 python3 /usr/bin/clickhouse-test --order=random --client-option http_make_head_request=0 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas=parallel_replicas parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200
5054 /bin/sh -c /usr/bin/clickhouse-test --order=random  --database=test_25  --client-option join_use_nulls=1 join_algorithm='full_sorting_merge' group_by_use_nulls=1 ignore_drop_queries_probability=0.5 --global_time_limit=1200  
5060 python3 /usr/bin/clickhouse-test --order=random --database=test_25 --client-option join_use_nulls=1 join_algorithm=full_sorting_merge group_by_use_nulls=1 ignore_drop_queries_probability=0.5 --global_time_limit=1200
5177 /bin/sh -c /usr/bin/clickhouse-test --order=random --db-engine="Replicated('/test/db/test_29', 's1', 'r1')"  --database=test_29  --client-option enable_deflate_qpl_codec=1 enable_zstd_qat_codec=1 join_algorithm='auto' max_rows_in_join=1000 group_by_use_nulls=1 ignore_drop_queries_probability=0.5 --global_time_limit=1200  
5180 python3 /usr/bin/clickhouse-test --order=random --db-engine=Replicated('/test/db/test_29', 's1', 'r1') --database=test_29 --client-option enable_deflate_qpl_codec=1 enable_zstd_qat_codec=1 join_algorithm=auto max_rows_in_join=1000 group_by_use_nulls=1 ignore_drop_queries_probability=0.5 --global_time_limit=1200
5261 /bin/sh -c /usr/bin/clickhouse-test --order=random  --database=test_31  --client-option join_use_nulls=1 join_algorithm='parallel_hash' memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 group_by_use_nulls=1 ignore_drop_queries_probability=0.5 --global_time_limit=1200  
5268 python3 /usr/bin/clickhouse-test --order=random --database=test_31 --client-option join_use_nulls=1 join_algorithm=parallel_hash memory_tracker_fault_probability=0.001 merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.05 group_by_use_nulls=1 ignore_drop_queries_probability=0.5 --global_time_limit=1200
5289 /bin/sh -c /usr/bin/clickhouse-test --order=random --db-engine="Replicated('/test/db/test_32', 's1', 'r1')"  --client-option enable_deflate_qpl_codec=1 enable_zstd_qat_codec=1 optimize_trivial_approximate_count_query=1 http_make_head_request=0 ignore_drop_queries_probability=0.5 --global_time_limit=1200  
5305 python3 /usr/bin/clickhouse-test --order=random --db-engine=Replicated('/test/db/test_32', 's1', 'r1') --client-option enable_deflate_qpl_codec=1 enable_zstd_qat_codec=1 optimize_trivial_approximate_count_query=1 http_make_head_request=0 ignore_drop_queries_probability=0.5 --global_time_limit=1200
5342 /bin/sh -c /usr/bin/clickhouse-test --order=random  --database=test_33  --client-option join_algorithm='partial_merge' group_by_use_nulls=1 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas='parallel_replicas' parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200  
5349 python3 /usr/bin/clickhouse-test --order=random --database=test_33 --client-option join_algorithm=partial_merge group_by_use_nulls=1 ignore_drop_queries_probability=0.5 allow_experimental_parallel_reading_from_replicas=1 max_parallel_replicas=3 cluster_for_parallel_replicas=parallel_replicas parallel_replicas_for_non_replicated_merge_tree=1 --global_time_limit=1200
Process group 1 should be killed
Traceback (most recent call last):
  File "/usr/bin/clickhouse-test", line 3599, in <module>
    main(args)
  File "/usr/bin/clickhouse-test", line 2957, in main
    total_tests_run += do_run_tests(args.jobs, test_suite)
  File "/usr/bin/clickhouse-test", line 2592, in do_run_tests
    run_tests_array(
  File "/usr/bin/clickhouse-test", line 2302, in run_tests_array
    stop_tests()
  File "/usr/bin/clickhouse-test", line 429, in stop_tests
    cleanup_child_processes(os.getpid())
  File "/usr/bin/clickhouse-test", line 417, in cleanup_child_processes
    child_pgid = os.getpgid(child)
ProcessLookupError: [Errno 3] No such process
```

Hence CI fails

azat committed:
tests: change the process group earlier to avoid killing self

Previously it was possible to have original pgid from the spawned
threads, that could lead to killing the caller script and in case of CI
it could be init process [1].

  [1]: https://s3.amazonaws.com/clickhouse-test-reports/67737/e68c9c8d16f37f6c25739076c9b071ed97952269/stress_test__asan_/stress_test_run_21.txt

Repro:

    $ echo "SELECT '1" > tests/queries/0_stateless/00001_select_1.sql # break the test
    $ cat /tmp/test.sh
    ./tests/clickhouse-test 0001_select --test-runs 3 --max-failures-chain 1 --no-random-settings --no-random-merge-tree-settings

Before this change:

    $ /tmp/test.sh
    Using queries from '/src/ch/worktrees/clickhouse-upstream/tests/queries' directory
    Connecting to ClickHouse server... OK
    Connected to server 24.8.1.1 @ bef896ce143ea4e0464c9829de6277ba06cc1a53 mt/rename-without-lock-v2
    Running 3 stateless tests (MainProcess).
    00001_select_1:                                                         [ FAIL ]
    Reason: return code:  62
    Code: 62. DB::Exception: Syntax error: failed at position 8 (''1;
    '): '1;
    . Single quoted string is not closed: ''1;
    '. (SYNTAX_ERROR)

    , result:

    stdout:

    Database: test_hz2zwymr
    Child processes of 13041:
    13042 python3 ./tests/clickhouse-test 0001_select --test-runs 3 --max-failures-chain 1 --no-random-settings --no-random-merge-tree-settings
    Killing process group 13040
    Processes in process group 13040:
    13040 -bash
    13042 python3 ./tests/clickhouse-test 0001_select --test-runs 3 --max-failures-chain 1 --no-random-settings --no-random-merge-tree-settings

    [2]+  Stopped                 /tmp/test.sh
    [1]$ Process group 13040 should be killed
    Max failures chain

    [2]+  Killed                  /tmp/test.sh

After:

    $ /tmp/test.sh
    Using queries from '/src/ch/worktrees/clickhouse-upstream/tests/queries' directory
    Connecting to ClickHouse server... OK
    Connected to server 24.8.1.1 @ bef896ce143ea4e0464c9829de6277ba06cc1a53 mt/rename-without-lock-v2
    Running 3 stateless tests (MainProcess).
    00001_select_1:                                                         [ FAIL ]
    Reason: return code:  62
    Code: 62. DB::Exception: Syntax error: failed at position 8 (''1;
    '): '1;
    . Single quoted string is not closed: ''1;
    '. (SYNTAX_ERROR)

    , result:

    stdout:

    Database: test_urz6rk5z
    Child processes of 9782:
    9785 python3 ./tests/clickhouse-test 0001_select --test-runs 3 --max-failures-chain 1 --no-random-settings --no-random-merge-tree-settings
    Max failures chain

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented:
>Stress test (ubsan) — Killed by signal (output files)

```
Has fatal logs from client:

########## Short fault info ############
(version 24.8.1.1501, build id: , git hash: ) (from thread 53132) Received signal 20
Signal description: Stopped
This is a signal used for debugging purposes by the user.
Stack trace: 0x00005586dbdd07ed 0x00005586dc201c38 0x00007f0f92e6d520 0x00005586dbd71c15 0x00005586dbd6d441 0x00005586dbd6ec0c 0x00005586dbd6d1c3 0x00005586dc0fa0c2 0x00005586dc0f43e8 0x00005586ee1d9b93 0x00005586dc10cdb1 0x00005586cf4dfbfe 0x00007f0f92e54d90 0x00007f0f92e54e40 0x00005586cf4c702e
########################################
(version 24.8.1.1501, build id: , git hash: ) (from thread 53132) (no query) Received signal Stopped (20)
This is a signal used for debugging purposes by the user.
Stack trace: 0x00005586dbdd07ed 0x00005586dc201c38 0x00007f0f92e6d520 0x00005586dbd71c15 0x00005586dbd6d441 0x00005586dbd6ec0c 0x00005586dbd6d1c3 0x00005586dc0fa0c2 0x00005586dc0f43e8 0x00005586ee1d9b93 0x00005586dc10cdb1 0x00005586cf4dfbfe 0x00007f0f92e54d90 0x00007f0f92e54e40 0x00005586cf4c702e
```

azat committed:
tests: increase delay co capture client stacktraces for sanitizers build

5 seconds is too small and not enough to print even few frames.

  [1]: https://s3.amazonaws.com/clickhouse-test-reports/67737/9658be5eea8351655dd3ea77b8c1d4717bac7999/stress_test__ubsan_.html

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
ci: use bash arrays to pass opts to clickhouse-test for stateless/stateful

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests: do not capture client stacktraces in stress tests

They are too uncontrollable, and likely will leave some clients [1].

  [1]: https://s3.amazonaws.com/clickhouse-test-reports/67737/9658be5eea8351655dd3ea77b8c1d4717bac7999/stress_test__ubsan_.html

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented:
Now it works as expected, CI failures does not related:

> AST fuzzer (debug) — Logical error: 'UNION query x AS (SELECT * FROM department__fuzz_0 UNION ALL SELECT * FROM q UNION ALL SELECT * FROM x) is not recursive'.

- https://github.com/ClickHouse/ClickHouse/issues/64571#issuecomment-2272822628

pamarcos commented on tests/clickhouse-test:
```diff
@@ -2353,7 +2408,13 @@ class BuildFlags:
 
 
+# Release and non-sanitizer build
+RELEASE_BUILD = False
```
I think naming the variable something more descriptive such as `RELEASE_NON_SANITIZED` or something along those lines helps understanding. The only implications of this var is the time we're sleeping. We could have at some point released but sanitized builds where you'd still want to sleep longer, right?

azat replied:
>I think naming the variable something more descriptive such as RELEASE_NON_SANITIZED or something along those lines helps understanding

OK, I don't have strong filling about it - will use your suggestiong

> We could have at some point released but sanitized builds where you'd still want to sleep longer, right?

I don't think so and I hope that there will be no more extra switches for sleeping, this will only make it less readable then

pamarcos replied:
No worries. I thought at first if it was called just `RELEASE` it'd rather be something along these lines, for consistency:

```python
RELEASE_BUILD = BuildFlags.RELEASE in result
```

But then when I realized the sleep amount it was probably due to either debug or because of sanitizers, I suggested the name change instead. It's not a major thing, in any case

pamarcos commented on tests/clickhouse-test:
```diff
@@ -3390,29 +3453,36 @@ def parse_args():
+    # Move to a new process group and kill it at exit so that we don't have any
+    # infinite tests processes left
+    # (new process group is required to avoid killing some parent processes)
+    os.setpgid(0, 0)
```
Good catch :ok_hand:

pamarcos commented on tests/clickhouse-test:
```diff
@@ -1248,39 +1325,35 @@ class TestCase:
-                    if e.errno != ESRCH:
-                        raise
-
                 if stderr:
```
Throwing a question to the air:

Are we not including in this branch `stdout` to `description` on purpose?
The inclusion of `stdout` seems missing here, while there are some redundant collections in the next branch (line 1395) and even some duplicated addition to `description` since it's in line 1413 but then also in 1441. Maybe this is the intending behavior, but I find it confusing

azat replied:
This code requires some cleanup.

> Are we not including in this branch stdout to description on purpose?

I don't think that `clickhouse-test` wrapper has some code that is "on purpose", but does not fills logical, and don't have comments for this "smart logic".

That said that I would say that 99% it is a bug in the code.

But let's address it separately, since the purpose of this PR is different.

>while there are some redundant collections in the next branch (line 1395) and even some duplicated addition to description since it's in line 1413 but then also in 1441. Maybe this is the intending behavior, but I find it confusing

About the case when stdout included twice - once if there is stderr, another if there is `@@SKIP@@` in stdout - I think it has been written with the assumption that SKIP cannot fail, and I guess it is "almost always" true

pamarcos reviewed (commented):
Very nice work here :superhero: 
I left some minor comments and questions

azat commented on tests/clickhouse-test:
```diff
@@ -3430,6 +3500,8 @@ if __name__ == "__main__":
         )
         sys.exit(1)
 
+    CAPTURE_CLIENT_STACKTRACE = args.capture_client_stacktrace
```
There is no scope, it not under function, just under `if`

pamarcos replied:
You're right. My bad :pray:

azat commented on tests/clickhouse-test:
```diff
@@ -267,7 +267,7 @@ def clickhouse_execute_http(
     retry_error_codes=False,
 ):
-    if args.secure:
+    if base_args.secure:
```
I guess the variable from `if __name__ == main` was used

azat committed:
tests/clickhouse-test: remove superior global

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented on tests/clickhouse-test:
```diff
@@ -358,14 +358,91 @@ def clickhouse_execute_json(
+
+
+def kill_process_group(pgid):
+    global CAPTURE_CLIENT_STACKTRACE
```
Thanks. Fixed

azat committed:
tests/clickhouse-test: update return type hint in run_single_test()

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests/clickhouse-test: better english in comment

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests/clickhouse-test: s/RELEASE_BUILD/RELEASE_NON_SANITIZED/g

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

nickitat commented:
after this pr we have number of failures with test_name = '['

https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZIb3VyKGNoZWNrX3N0YXJ0X3RpbWUpIGFzIGQsCmNvdW50KCksIGdyb3VwVW5pcUFycmF5KHB1bGxfcmVxdWVzdF9udW1iZXIpLCAgYW55KHJlcG9ydF91cmwpLCBncm91cFVuaXFBcnJheSh0ZXN0X25hbWUpCmZyb20gY2hlY2tzIHdoZXJlIHRvZGF5KCkgLSBJTlRFUlZBTCAnMyBtb250aHMnIDw9IGNoZWNrX3N0YXJ0X3RpbWUgYW5kIHRlc3RfbmFtZSBsaWtlICdbJyBhbmQgdGVzdF9zdGF0dXMgaW4gKCdGQUlMJywgJ0ZMQUtZJykgZ3JvdXAgYnkgZCBvcmRlciBieSBkIGRlc2M=

azat commented:
Yeah, known issue, should be fixed by https://github.com/ClickHouse/ClickHouse/pull/68487
